spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -1632,6 +1633,85 @@ struct ProductionVisitor
             parseProduction(i);
     }
 
+    void operator()(const production::Skip& p) {
+        if ( const auto& ctor = p.ctor() ){
+            pb->skipLiteral(*ctor);
+        }
+
+        else if ( p.field().parseType().isA<type::Bytes>() ) {
+            auto eod_attr = AttributeSet::find(p.field().attributes(), "&eod");
+            auto size_attr = AttributeSet::find(p.field().attributes(), "&size");
+            auto until_attr = AttributeSet::find(p.field().attributes(), "&until");
+            if ( ! until_attr )
+                until_attr = AttributeSet::find(p.field().attributes(), "&until-including");
+
+            if ( auto c = p.field().condition() )
+                pushBuilder(builder()->addIf(*c));
+
+            if ( size_attr ) {
+                auto n = builder()->addTmp("skip", *size_attr->valueAsExpression());
+                auto loop = builder()->addWhile(builder::greater(n, builder::integer(0U)));
+                pushBuilder(loop, [&]() {
+                    pb->waitForInput(builder::integer(1U), "not enough bytes for skipping", p.location());
+                    auto consume = builder()->addTmp("consume", builder::min(builder::size(state().cur),
+                                                                             *size_attr->valueAsExpression()));
+                    pb->advanceInput(consume);
+                    builder()->addAssign(n, builder::difference(n, consume));
+                    builder()->addDebugMsg("spicy-verbose", "- skipped %u bytes (%u left to skip)", {consume, n});
+                });
+            }
+
+            else if ( eod_attr ) {
+                builder()->addDebugMsg("spicy-verbose", "- skipping to eod");
+                pb->waitForEod();
+                pb->advanceInput(builder::size(state().cur));
```

Comment:
This comes straight from the existing production for `void &eod`, but as a comment, we are leaving some performance on the table here by keeping all data until EOD in the stream before attempting to trim, e.g., for a field `nope: skip bytes &eod` we generate

```python
# Begin parsing production: Skip: nope -> skip: nope
spicy_rt::waitForEod(__data, __cur, Null);
__cur = __cur.advance(|__cur|);

if ( __trim ) 
    (*__data).trim(begin(__cur));

(*self).__error = __error;
default<void>();
__error = (*self).__error;
# End parsing production: Skip: nope -> skip: nope
```

We couldn't really do much better before, but with `skip` we have a chance. We could e.g., use `waitForInputOrEod` in a loop here and trim on each iteration.

Reply:
Good catch, will change.